app/wiring/AppLoader.scala

```diff
@@ -32,10 +33,20 @@ class AppLoader extends ApplicationLoader {
         "DEV" -> new AppComponents(AppComponents.DEV, this, controllerComponents, this).router
       )
 
-      lazy val defaultRouter = applicationComponentsMap("PROD")
+      lazy val router = applicationComponentsMap("PROD")
```

Comment:
overriding `handlerFor` in Router wasn't allowed and had been made final, so I found the equivalent in play 2.7 docs and implemented that: https://www.playframework.com/documentation/2.7.x/ScalaHttpRequestHandlers

Reply:
From the introduction paragraphs of the same link

> Providing a custom HttpRequestHandler **should be a last course of action**.

Is there no other way to define 

```
lazy val membershipRatePlanIds = ???
lazy val digipackRatePlanIds = ???
```

Reply:
true I did see that and it makes sense for something new, but my reasoning was we were moving from something totally unsupported (overriding handlerFor I couldn't see any documentation allowing it) to something not recommended.
Trying to modify the AppComponents would be a more intrusive change, and the "stage" seems to get passed to a lot of stuff in that appComponents, even though it only seems to use the `.router` from that.
Getting off the unsupported play is relatively important so I didn't spend a long time refactoring.  Although if you have a bit of time to dedicate it would be a good idea to do it a more proper way!